feat: Added Netlify as a provider. (#1052)
qdm12/ddns-updater: docs/netlify.md

# Netlify

## Configuration

### Example

```json
{
  "settings": [
    {
      "provider": "netlify",
      "domain": "domain.com",
      "token": "your-netlify-access-token",
      "ip_version": "ipv4",
      "ipv6_suffix": ""
    }
  ]
}
```

### Compulsory parameters

- `"domain"` is the domain to update. It can be `example.com` (root domain), `sub.example.com` (subdomain of `example.com`) or `*.example.com` for wildcard.
- `"token"` is your Netlify personal access token with DNS zone permissions

### Optional parameters

- `"ip_version"` can be `ipv4` (A records), or `ipv6` (AAAA records) or `ipv4 or ipv6` (update one of the two, depending on the public IP found). It defaults to `ipv4 or ipv6`.
- `"ipv6_suffix"` is IPv6 interface identifier suffix to use. It can be for example `0:0:0:0:72ad:8fbb:a54e:bedd/64`. If left empty, it defaults to no suffix and the raw public IPv6 address obtained is used in record updating.

## Domain setup

[![Netlify Website](../readme/netlify.png)](https://www.netlify.com)

1. Login to your Netlify account at [https://app.netlify.com/](https://app.netlify.com/)

> 
> 2. Navigate to **Site settings** for your site
> 
> 3. Go to **Domain management** → **DNS zones**
> 
> 4. Ensure your domain is properly configured as a DNS zone in Netlify

Reviewer: nit let's use `1.` everywhere so github markdown handles the numbering

```suggestion
1. Navigate to **Site settings** for your site
1. Go to **Domain management** → **DNS zones**
1. Ensure your domain is properly configured as a DNS zone in Netlify
```

> 
> ## Token setup
> 

Reviewer: let's just continue directly to the token setup

```suggestion
```

1. Go to **User settings** → **Applications** → **Personal access tokens**

> 
> 2. Click **New access token**
> 
> 3. Give the token a descriptive name (e.g., "DDNS Updater")
> 
> 4. Select the following scopes:
>    - **DNS:read** - Read DNS zones and records
>    - **DNS:edit** - Edit DNS zones and records
> 
> 5. Click **Generate token**
> 
> 6. Copy the generated token - this is your `"token"` value

Reviewer: ```suggestion
1. Click **New access token**
1. Give the token a descriptive name (e.g., "DDNS Updater")
1. Select the following scopes:
   - **DNS:read** - Read DNS zones and records
   - **DNS:edit** - Edit DNS zones and records
1. Click **Generate token** - this is your `"token"` value
```

> ## Testing
> 
> 1. Go to your Netlify site's DNS management page
> 
> 2. Check the current DNS record for your domain
> 
> 3. Run ddns-updater
> 
> 4. Refresh the Netlify DNS page to verify the update occurred

Reviewer: I appreciate the effort in documentation, but let's remove this section, I don't think it's really necessary. ddns-updater should log if anything wrong happens.

```suggestion
```

## Notes

> - Netlify's DNS API requires the domain to be configured as a DNS zone in your Netlify account

Reviewer: this is already mentioned in the domain setup section

```suggestion
```

> - The provider automatically finds the appropriate DNS zone for your domain

Reviewer: what does that mean?

> - If a DNS record doesn't exist, it will be created

Reviewer: this is expected default behavior (except with a few providers where it's not possible) - let's remove

```suggestion
```

> - If a DNS record exists with a different IP, it will be deleted and recreated with the new IP

Reviewer: that feels like an implementation detail, no need to document it

```suggestion
```

> - The default TTL for created records is 3600 seconds (1 hour)

Reviewer: this should be an optional `"ttl"` parameter, like with other providers. You can mention in defaults to 1hr in the optional parameters section above, and that should also be removed.

```suggestion
```

> - IPv6 is supported if your Netlify account has IPv6 enabled for the DNS zone

Reviewer: Should this be moved to the domain setup section? And when you say "IPv6 enabled", what does that mean exactly? Is there a setting/tickbox on netlify's UI?
